test_helpers/integration_helpers_test.go

```diff
@@ -0,0 +1,123 @@
+package test_helpers_test
+
+import (
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testhelpers "github.com/maximilien/bosh-softlayer-cpi/test_helpers"
+)
+
+var _ = Describe("helper functions for integration tests", func() {
+	var (
+		cpiTemplate testhelpers.CpiTemplate
+
+		replacementMap map[string]string
+
+		rootTemplatePath string
+	)
+
+	BeforeEach(func() {
+		replacementMap = map[string]string{
+			"ID":           "some ID",
+			"DirectorUuid": "some director UUID",
+			"Tags":         "some, tags",
+		}
+
+		pwd, err := os.Getwd()
+		Expect(err).ToNot(HaveOccurred())
+		rootTemplatePath = filepath.Join(pwd, "..")
+	})
+
+	Context("#RunCpi", func() {
+		It("/out/cpi to exist and run", func() {
+			configPath := filepath.Join(rootTemplatePath, "dev", "config.json")
+			payload := `{
+							"method": "set_vm_metadata",
+							"arguments": [
+								"some ID",
+								{
+									"director": "BOSH Director",
+									"deployment": "softlayer",
+									"tags": "some, tags"
+								}
+							],
+							"context": {
+								"director_uuid": "some director UUID"
+							}
+						}`
+			err := testhelpers.RunCpi(rootTemplatePath, configPath, payload)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+	})
+
+	Context("#CreateTmpConfigPath", func() {
+		It("creates a config.json in temp dir with new config template values", func() {
+			tmpConfigPath, err := testhelpers.CreateTmpConfigPath(rootTemplatePath, "test_fixtures/cpi_methods/config.json", "some username", "some ApiKey")
+			Expect(err).ToNot(HaveOccurred())
+
+			fileInfo, err := os.Stat(tmpConfigPath)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(fileInfo.Mode().IsRegular()).To(BeTrue())
+
+			err = os.RemoveAll(tmpConfigPath)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("#GenerateCpiJsonPayload", func() {
+		//TODO: Context("error cases", ...)
```

Comment:
Let's either implement these tests now or remove the TODO and move them as one issue in the repo and chore on Tracker.